app/src/main/java/com/simon/harmonichackernews/UserDialogFragment.java

```diff
@@ -106,7 +106,15 @@ public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
                             int karma = jsonObject.getInt("karma");
                             Calendar cal = Calendar.getInstance();
                             cal.setTime(new Date(jsonObject.getInt("created") * 1000L));
-                            JSONArray submitted = jsonObject.getJSONArray("submitted");
+
+                            // Users who have never submitted before do not receive this key as part of their response.
+                            JSONArray submitted = null;
+                            if (jsonObject.has("submitted")) {
+                                submitted = jsonObject.getJSONArray("submitted");
+                            } else {
+                                submitted = new JSONArray();
+                                jsonObject.put("submitted", submitted);
+                            }
```

Comment:
I just defaulted this to an empty array, but if you'd prefer we could instead add a `null` check below on line 119.

Reply:
I think this behavior with an empty array is less likely to cause issues - this bring up the idea that the submissions button should probably be hidden from the dialog if the 'submitted' array is empty